### PR TITLE
[REG-1874] Support games whose active input handling is set to exclusively use the new Input System

### PR DESCRIPTION
**Pull request by svolokh**
This PR addresses an issue where if Build Settings -> Player Settings -> Active Input Handling is set to “Input System Package (New)”, rather than “Both”, the SDK currently errors out due to a variety of dependency on both the old/new input systems throughout the SDK. This adds ENABLE_LEGACY_INPUT_MANAGER checks in all the remaining places where they should be to disable the legacy input usage when the game is exclusively using the new Input System.

As before, it is out of scope to fully remove dependency on the new Input System in the SDK for games that only use the legacy input system. This PR is just to ensure that the configurations of "Both" and "Input System Package (New)" are both working.

This was tested in two ways:
1. The unit tests in the RGUnityBots project were run with active input handling set to New Input System only.
2. Boss Room was re-configured to be set to use the new Input System only, and the record/replay and monkey bot were checked to work as expected.

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs [hunk 1]
```diff
@@ -253,7 +257,7 @@ private IEnumerable<ExpressionSyntax> FindCandidateValuesForLocalVariable(ILocal
         /// </summary>
         private IEnumerable<SyntaxNode> FindDerivedUserInput(ExpressionSyntax expr)
         {
-            const int maxDepth = 5;
+            const int maxDepth = 10;
```
Comment (svolokh) [inline src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs:260]: This change (and the other one in this method) is to fix a bug I found with the analysis of raycasts when checking raycast from the new Input System's mouse position.

**Review by RG-nAmKcAz — APPROVED** [review]

**Review by vontell — APPROVED** [review]
